src/wp-includes/class-wp-html-attribute-token.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * HTML Tag Processor: Attribute token structure class.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ * @since 6.2.0
+ */
+
+if ( ! class_exists( 'WP_HTML_Attribute_Token' ) ) :
```

Comment:
We should remove these guards, see https://github.com/WordPress/wordpress-develop/pull/3920#issuecomment-1408636103 😊 (Same for the other classes' guards.)

Reply:
Yes, it's only necessary in the Gutenberg plugin.

Reply:
I agree. 

>Core is the most "canonical" source for those classes, and its versions should "prevail"

Core does not have the responsibility of checking if its classes is already in memory. It's the responsibility of code extending from Core, i.e. plugins and themes. Also noting, Core's classes load into memory before plugins or themes.

Yes, I agree these `class_exists()` guards should be removed.

Reply:
Noted in the trac ticket, I have to say I want to disagree. Maybe I've dealt with too many bugs that were there because someone thought they needed to `require` a file they didn't need to. Also this is an established pattern in Core: what's different about this one that we ought to remove the safeties that keep a site from crashing?

Reply:
>Also this is an established pattern in Core: what's different about this one that we ought to remove the safeties that keep a site from crashing?

@dmsnell take a look at https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-settings.php. Notice each of `require` does not first check if that class is already loaded into memory before Core loads its class file.

Reply:
Also Core loads API files into memory within `wp-settings.php`.  For consistency, I moved the loading there and removed the `class_exists()` wrappers https://github.com/WordPress/wordpress-develop/commit/b708c6b618770c7e80981c93c0bd2ab9d73a7819.

Reply:
Linking the conversations here for backscrolling and context:

https://github.com/WordPress/wordpress-develop/pull/3920#issuecomment-1409141336

https://github.com/WordPress/wordpress-develop/pull/3920#issuecomment-1409568369

Reply:
Noting for cross-referencing that this has been brought up again in https://github.com/WordPress/gutenberg/issues/58467